### PR TITLE
Dark mode and servlet fetching

### PR DESCRIPTION
**Pull request by azhara-a**
**Implement dark mode**
Screenshot 2020-07-20 at 10 33 31 AM
Screenshot 2020-07-20 at 10 33 44 AM

**Add a button that fetches fun facts from Data Servlet** 
Screenshot 2020-07-20 at 1 46 48 PM

### CODE REVIEW

File: portfolio/src/main/webapp/dark-mode-switch.min.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const darkSwitch = document.getElementById("darkSwitch");
+
+function initTheme() {
+    const e = null !== localStorage.getItem("darkSwitch") &&
```
Comment (mugdhalakhani) [inline portfolio/src/main/webapp/dark-mode-switch.min.js:4]: nit: give this a descriptive name.

Comment (azhara-a) [reply portfolio/src/main/webapp/dark-mode-switch.min.js:4]: Is retrievePreviousTheme more descriptive?

Comment (azhara-a) [reply portfolio/src/main/webapp/dark-mode-switch.min.js:4]: I changed the name in the next commit

File: portfolio/src/main/webapp/dark-mode.css [hunk 2]
```diff
@@ -0,0 +1,42 @@
+[data-theme="dark"] {
+    background-color: #121212 !important;
```
[the remark this answers is not in the document]
Comment (azhara-a) [reply portfolio/src/main/webapp/dark-mode.css:2]: For the light theme, I used !important annotations to change the default colours of the bootstrap components. I will look into finding a way to remove !important

File: portfolio/src/main/webapp/dark-mode-switch.min.js (hunk 1 above)
Comment (mugdhalakhani) [reply portfolio/src/main/webapp/dark-mode-switch.min.js:4]: My comment was about the variable 'e'. It's useful to give variables a descriptive name, improves readability of the code.

Comment (azhara-a) [reply portfolio/src/main/webapp/dark-mode-switch.min.js:4]: Sorry, I misunderstood

File: portfolio/src/main/webapp/dark-mode.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (mugdhalakhani) [reply portfolio/src/main/webapp/dark-mode.css]: thanks for removing these annotations!

**Pull request merged**